services/site/tests/components/challenge/level/QuizLevel.spec.tsx

```diff
@@ -70,79 +70,101 @@ const mocks: MockedResponse[] = [
   },
 ];
 
+const mockedResponsesForIncorrect: MockedResponse[] = [
+  {
+    request: {
+      query: SubmitQuizLevelAnswerDocument,
+      variables: {
+        input: {
+          levelId: "1",
+          answers: ["1"],
+        },
+      },
+    },
+    result: {
+      data: {
+        submitQuizLevelAnswer: {
+          result: { id: "1", status: ResultStatus.Fail },
+        },
+      },
+    },
+  },
+];
+
 beforeEach(() => {
   jest.resetAllMocks();
   router.query = { challengeSlug: mockChallengeSlug, nthLevel: String(mockNthLevel) };
   router.back = jest.fn();
 });
 
 describe("Quiz Level", () => {
-  it("renders wrapper elements", () => {
-    const wrapper = mount(
-      <MockedProvider mocks={mocks}>
-        <QuizLevel question={mockText} answers={mockAnswers} isLastLevel={false} levelId={"1"} />
-      </MockedProvider>,
-    );
-
-    expect(wrapper.exists("section")).toBeTruthy();
-  });
-
   it("renders heading", () => {
-    const wrapper = mount(
-      <MockedProvider mocks={mocks}>
+    render(
+      <MockedProvider mocks={mockedResponsesForSuccess}>
         <QuizLevel question={mockText} answers={mockAnswers} isLastLevel={false} levelId={"1"} />
       </MockedProvider>,
     );
 
-    expect(wrapper.exists("h2")).toBeTruthy();
-    expect(wrapper.find("h2").text()).toBe(mockHeading);
+    expect(screen.getByRole("heading", { level: 2, name: mockHeading })).toBeInTheDocument();
   });
 
   it("renders quiz question as html", () => {
-    const wrapper = mount(
-      <MockedProvider mocks={mocks}>
+    render(
+      <MockedProvider mocks={mockedResponsesForSuccess}>
         <QuizLevel question={mockText} answers={mockAnswers} isLastLevel={false} levelId={"1"} />
       </MockedProvider>,
     );
 
-    expect(wrapper.find("h3").html()).toContain(mockText);
+    expect(screen.getByRole("heading", { level: 3 }).innerHTML).toContain(mockText);
   });
 
   it("renders quiz answers", () => {
-    const wrapper = mount(
-      <MockedProvider mocks={mocks}>
+    render(
+      <MockedProvider mocks={mockedResponsesForSuccess}>
         <QuizLevel question={mockText} answers={mockAnswers} isLastLevel={false} levelId={"1"} />
       </MockedProvider>,
     );
 
-    expect(wrapper.exists(SingleAnswer)).toBeTruthy();
+    expect(screen.getByText(mockAnswers.at(0).text)).toBeInTheDocument();
   });
 
-  it("renders fail animation when quiz status is fail", () => {
-    const wrapper = mount(
-      <MockedProvider mocks={mocks}>
+  it("renders fail animation when quiz status is fail", async () => {
+    render(
+      <MockedProvider mocks={mockedResponsesForSuccess}>
         <QuizLevel question={mockText} answers={mockAnswers} isLastLevel={false} levelId={"1"} />
       </MockedProvider>,
     );
 
-    expect(wrapper.find(Lottie).length).toBe(0);
```

Comment:
maybe I just don't understand it correctly, but how do you check if the Lottie animation is present now?

Reply:
At the top of the file we define a mock for the Lottie component that is rendered instead of the actual Lottie animation. That component contains a `data-testid="lottie"` attribute. If we can find an element with that test id (done via `.getByTestId("lottie")`, we can assume that the animation will be rendered in the actual application (instead of the mock).
I hope that somewhat explains it. If not, let me know.